datafusion/sql/src/unparser/utils.rs

```diff
@@ -85,6 +85,69 @@ pub(crate) fn find_unnest_node_within_select(
     }
 }
 
+pub (crate) fn try_transform_to_simple_table_scan_with_filters(plan: &LogicalPlan) -> Option<(LogicalPlan, Vec<Expr>)> { 
```

Comment:
Add comments explaining what this function does

Reply:
Added - thank you

```
/// Iterates through the children of a [LogicalPlan] to find a TableScan node before encountering
/// a Projection or any unexpected node that indicates the presence of a Projection (SELECT) in the plan.
/// If a TableScan node is found, returns the TableScan node without filters, along with the collected filters separately.
/// If the plan contains a Projection, returns None.
/// 
/// Note: If a table alias is present, TableScan filters are rewritten to reference the alias.
///
/// LogicalPlan example:
///   Filter: ta.j1_id < 5 
///     Alias:  ta
///       TableScan: j1, j1_id > 10 
///
/// Will return LogicalPlan below:
///     Alias:  ta
///       TableScan: j1
/// And filters: [ta.j1_id < 5, ta.j1_id > 10]
```